### PR TITLE
curl: add --no-clobber

### PR DESCRIPTION
**Pull request by Hex052**
I have no idea how to write a test that would cover this, if someone could help me with writing some, I'll happily add them!

This is something from the TODO list:
> 18.3 prevent file overwriting
>
> Add an option that prevents curl from overwriting existing local files. When used, and there already is an existing file with the target file name (either -O or -o), a number should be appended (and increased if already existing). So that index.html becomes first index.html.1 and then index.html.2 etc.

### CODE REVIEW

**Comment by Hex052** [comment]
@jay could I get some help writing some unit tests for this? Sorry if you're the wrong person to ask!

**Comment by jay** [comment]
FYI it looks like this was attempted once in 2014
https://curl.se/mail/archive-2014-05/0014.html

File: docs/cmdline-opts/no-clobber.d [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Long: no-clobber
+Help: Create necessary local directory hierarchy
```
Comment (bagder) [inline docs/cmdline-opts/no-clobber.d:2]: That seems like a wrong description!

File: docs/options-in-versions [hunk 2]
```diff
@@ -125,6 +125,7 @@
 --next (-:)                          7.36.0
 --no-alpn                            7.36.0
 --no-buffer (-N)                     6.5
+--no-clobber                         7.79.0
```
Comment (bagder) [inline docs/options-in-versions:128]: 7.80.0 is now the next opportunity

File: src/tool_help.c [hunk 3]
```diff
@@ -850,6 +850,12 @@ static const struct helptxt helptext[] = {
   {"    --xattr",
    "Store metadata in extended file attributes",
    CURLHELP_MISC},
+  {"    --clobber",
+   "Overwrite files that already exist, even for options that normally don't",
+   CURLHELP_OUTPUT},
+  {"    --no-clobber",
+   "Never overwrite files that already exist, instead appending .1 - .100",
+   CURLHELP_OUTPUT},
```
Comment (bagder) [inline src/tool_help.c:858]: Only document the `--no-clobber` one, and note that this list is (should be) generated - as documented at the top of it.

File: docs/cmdline-opts/no-clobber.d (hunk 1 above)
Comment (Hex052) [reply docs/cmdline-opts/no-clobber.d:2]: Whoops! Guess you can tell which one I used as a template

File: docs/options-in-versions (hunk 2 above)
Comment (Hex052) [reply docs/options-in-versions:128]: Changed

**Comment by bagder** [comment]
@Hex052 can you please fix the merge issue and force-push an update? Then I can have a look at making a test case or two for this!

**Comment by bagder** [comment]
I rebased, squashed and took the liberty to force-push.

**Comment by bagder** [comment]
@Hex052 I'm getting close to merging this PR. Do you want to be credited as **HexTheDragon** for this change?

**Comment by bagder** [comment]
Thanks!

**Comment by Hex052** [comment]
> @Hex052 I'm getting close to merging this PR. Do you want to be credited as **HexTheDragon** for this change?

Sure, that works!

**Pull request closed without merging**